### PR TITLE
add optional TLS client auth to Kafka storage

### PR DESCRIPTION
**Pull request by devx**
This pull requests adds the optional TLS client authentication to the kafka storage driver.

### CODE REVIEW

File: docs/kafka.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+```
+-storage_driver_kafka_topic=myTopic
+```
+
+Specify a kafka topic:
+
+```
+-storage_driver_kafka_topic=myTopic
+```
```
Comment (timstclair) [inline docs/kafka.md]: Duplicated

File: docs/kafka.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# Exporting cAdvisor Stats to Kafka
+
+cAdvisor supports exporting stats to [Kafka](http://kafka.apache.org/). To use kafka, you need to provide the additional flags to cAdvisor:
+
+Set the storage driver as kafka:
```
Comment (timstclair) [inline docs/kafka.md]: (nit) Capitalize Kafka everywhere

File: storage/kafka/kafka.go [hunk 3]
```diff
@@ -34,8 +37,12 @@ func init() {
...
 
 var (
-	brokers = flag.String("storage_driver_kafka_broker_list", "localhost:9092", "kafka broker(s) csv")
-	topic   = flag.String("storage_driver_kafka_topic", "stats", "kafka topic")
+	brokers   = flag.String("storage_driver_kafka_broker_list", "localhost:9092", "kafka broker(s) csv")
+	topic     = flag.String("storage_driver_kafka_topic", "stats", "kafka topic")
+	certFile  = flag.String("storage_driver_kafka_cert", "", "optional certificate file for TLS client authentication")
+	keyFile   = flag.String("storage_driver_kafka_key", "", "optional key file for TLS client authentication")
+	caFile    = flag.String("storage_driver_kafka_ca", "", "optional certificate authority file for TLS client authentication")
+	verifySSL = flag.Bool("storage_driver_kafka_verifySSL", true, "verify ssl certificate chain")
```
Comment (timstclair) [inline storage/kafka/kafka.go]: optional: consider simplifying the flag name to `storage_driver_kafka_ssl`

File: storage/kafka/kafka.go [hunk 4]
```diff
@@ -94,8 +101,38 @@ func new() (storage.StorageDriver, error) {
 	return newStorage(machineName)
 }
 
+func generateTLSConfig() *tls.Config {
+	if *certFile != "" && *keyFile != "" && *caFile != "" {
+		cert, err := tls.LoadX509KeyPair(*certFile, *keyFile)
+		if err != nil {
+			glog.Fatal(err)
```
Comment (timstclair) [inline storage/kafka/kafka.go]: This should return the error instead.

**Comment by timstclair** [comment]
ok to test

**Comment by devx** [comment]
I have made the suggested changes.

-Victor

File: storage/kafka/kafka.go [hunk 5]
```diff
@@ -94,8 +101,43 @@ func new() (storage.StorageDriver, error) {
...
+
+	return nil, nil
+}
+
 func newStorage(machineName string) (storage.StorageDriver, error) {
 	config := kafka.NewConfig()
+
+	tlsConfig, err := generateTLSConfig()
+	if err != nil {
+		glog.Fatalf("Invalid Kafka SSL configuration: %s", err)
```
Comment (timstclair) [inline storage/kafka/kafka.go]: Return the error here too.

**Comment by timstclair** [comment]
Thanks for the documentation, could you please sync it with the structure that was just merged in https://github.com/google/cadvisor/pull/1265 (move to docs/storage/kafka.md, and add a link to docs/storage/README.md). Everything else LGTM.

**Comment by devx** [comment]
Documentation updated.  Thank you for the guidance, and patience.

**Comment by timstclair** [comment]
LGTM

**Pull request merged**